src/lsdb/catalog/dataset/healpix_dataset.py

```diff
@@ -408,10 +409,9 @@ def plot_pixels(self, projection: str = "moll", **kwargs):
 
     def to_hipscat(
         self,
-        base_catalog_path: str,
+        base_catalog_path: UPath,
```

Comment:
Can we have the user facing functions as `UPath | str`? Like this and `read_hipscat`

Reply:
Switched a bunch of places to use `str | Path | UPath`.